pkg/server/commands/connectedobjects/connected_objects.go

```diff
@@ -161,15 +162,17 @@ func (c *ConnectedObjectsQuery) Execute(
 	ctx context.Context,
 	req *ConnectedObjectsRequest,
 	resultChan chan<- *ConnectedObjectsResult,
+	dsQueryCount *uint32,
```

Comment:
Instead of passing this around everywhere, why don't we do something similar to what we did in Check and propagate some ResolutionMetadata in each ConnectedObjectsRequest? Then for each object we find we can include the resolution metadata in the ConnectedObjectsResult.

I'd prefer to encapsulate the resolution metadata as part of the request body because it avoids bloating the signature, is more consistent with code elsewhere, and it is more flexible in that it potentially allows us to distribute the ConnectedObjects expansion in a distributed way via dispatching. The way this is written right now doesn't allow for a distributed processing approach down since it assumes a single host is handling the request and makes use of shared memory between recursive calls. If we include this metadata in the request structure then we achieve the same thing but do so in a way that is more flexible.

Reply:
> Instead of passing this around everywhere, why don't we do something similar to what we did in Check and propagate some ResolutionMetadata in each ConnectedObjectsRequest? Then for each object we find we can include the resolution metadata in the ConnectedObjectsResult.
> 
> I'd prefer to encapsulate the resolution metadata as part of the request body because it avoids bloating the signature, is more consistent with code elsewhere, and it is more flexible in that it potentially allows us to distribute the ConnectedObjects expansion in a distributed way via dispatching. The way this is written right now doesn't allow for a distributed processing approach since it assumes a single host is handling the request and makes use of shared memory between recursive calls. If we include this metadata in the request structure then we achieve the same thing but do so in a way that is more flexible.

The problem is that the datastore query call should be per request rather than per returned result.  Otherwise, we may overcount as pointed out by @miparnisari  in https://github.com/openfga/openfga/pull/959#discussion_r1303577444

Reply:
As discussed offline, we should use ResolutionMetadata instead of dsQueryCount to allow future expansion without bloating the signature.